js/src/provider/SubscriptionManager.ts

```diff
@@ -0,0 +1,66 @@
+import SafeEventEmitter from "@metamask/safe-event-emitter"
+import { RequestArguments, Web3Provider } from "./Web3Provider"
+import {
+  JsonRpcEngineNextCallback,
+  JsonRpcEngineEndCallback
+} from "json-rpc-engine"
+const PollingBlockTracker = require("eth-block-tracker")
+const createSubscriptionManager = require("eth-json-rpc-filters/subscriptionManager")
+const noop = () => {}
+
+export interface SubscriptionResult {
+  result?: unknown
+}
+
+export interface SubscriptionNotification {
+  method: string
+  params: {
+    subscription: string
+    result: unknown
+  }
+}
+
+export class SubscriptionManager {
```

Comment:
Please add class-level doc -- might be worth linking to https://eips.ethereum.org/EIPS/eip-1193